pkg/skaffold/runner/dev.go

```diff
@@ -79,8 +83,7 @@ func (r *SkaffoldRunner) Dev(ctx context.Context, out io.Writer, artifacts []*la
 	r.createLogger(out, artifacts)
 	defer r.logger.Stop()
 
-	forwarderManager := portforward.NewForwarderManager(out, r.imageList, r.runCtx.Namespaces, r.defaultLabeller.K8sManagedByLabelKeyValueString(), r.runCtx.Opts.PortForward, r.portForwardResources)
-	defer forwarderManager.Stop()
```

Comment:
Why not defer the stop anymore?

Reply:
So I think having the stop here is actually a bug. The stop should happen at the end of every dev loop, not at the end of the dev command, since we want to clean up ports and port-forward again on every cycle.

Reply:
It's a slightly related problem? with Nick's PR #2501 the config change scenario is back in action. And I just tried it and because of removing this `Stop` here, we leave kubectl processes hanging around. We could add the stop in the devLoop before returning the error though.

Reply:
I added the `Stop` before throwing the `ErrorConfigurationChanged` as well.